### ISSUE TITLE
Issue decoding XML character entity reference

### ISSUE THREAD

**Issue by kamcma** [sersoft-gmbh/xmlwrangler#120]
Hello,

I run into an issue when attempting to decode an XML character entity reference. The space after an XML character entity reference is dropped when I think it should not be. I have cooked up a simple test to demonstrate the issue.
```swift
func testXMLStringDecoding() throws {
  let xml = """
  <?xml version="1.0" encoding="UTF-8"?>
  <root myattr="myvalue">
      <child1>One &amp; two</child1>
  </root>
  """

  let root = try XMLElement.parse(xml)
  let content = try root.content.findFirst(elementNamed: "child1")?.stringContent()
  
  XCTAssertEqual("One & two", content) // 🛑 XCTAssertEqual failed: ("Optional("One & two")") is not equal to ("Optional("One &two")")
}
```

Demo Swift package with the above failing test case.
[xml-string-decoding-demo.zip](https://github.com/sersoft-gmbh/xmlwrangler/files/14672979/xml-string-decoding-demo.zip)

**Comment by ffried** [comment]
Great catch! And also an awesome bug report, including a test case!

Foundation's `XMLParser` breaks text containing XML character entity references into pieces and delivers each piece to the delegate: `"One "`, `"&"`, and `" two"`. This led to the piece being stripped of whitespaces (which is done to correctly deal with multiline string content), thus removing the leading whitespace of `" two"` (the first part is treated slightly differently)...

**Comment by ffried** [comment]
This is now fixed and released in [v6.1.1](https://github.com/sersoft-gmbh/xmlwrangler/releases/tag/6.1.1).

**Comment by kamcma** [comment]
Wow, impressive speed! Thank you!
